packages/data/src/components/use-dispatch/test/use-dispatch.js

```diff
@@ -11,165 +12,123 @@ import createReduxStore from '../../../redux-store';
 import { createRegistry } from '../../../registry';
 import { RegistryProvider } from '../../registry-provider';
 
+jest.useRealTimers();
+
 describe( 'useDispatch', () => {
+	const counterStore = {
+		reducer: ( state = 0, action ) => {
+			if ( action.type === 'INC' ) {
+				return state + 1;
+			}
+
+			return state;
+		},
+		actions: {
+			inc: () => ( { type: 'INC' } ),
+		},
+		selectors: {
+			get: ( state ) => state,
+		},
+	};
+
 	let registry;
 	beforeEach( () => {
 		registry = createRegistry();
 	} );
 
-	it( 'returns dispatch function from store with no store name provided', () => {
-		registry.registerStore( 'demoStore', {
-			reducer: ( state ) => state,
-			actions: {
-				foo: () => 'bar',
-			},
-		} );
+	it( 'returns dispatch function from store with no store name provided', async () => {
+		const user = userEvent.setup();
```

Comment:
For consistency with the rest of the tests, let's make sure to initiate user-event like this:

```suggestion
		const user = userEvent.setup( {
			advanceTimers: jest.advanceTimersByTime,
		} );
```

Reply:
I've no strong preference here. Using `jest.useRealTimers()` was suggested above - https://github.com/WordPress/gutenberg/pull/44802#discussion_r991054058.

Reply:
My suggestion is mostly to keep consistency with the rest of the tests throughout the codebase, where we use `jest.advanceTimersByTime()`. It is kind of predictable to know that timers work the same way across all tests 😉 No strong feelings towards it, either, I just recall @ciampo suggesting we keep consistency with this and I always favor consistency when possible.

Reply:
I think the test should choose between fake/real timers primarily according to what it tests and what timers it needs. Consistency is secondary.

The actual tested code doesn't use any timers at all. It's just that `userEvent.click` internally dispatches two events, "move mouse to the element" and "click on the element," with a zero delay between them (`setTimeout( x, 0 )`). So it has to deal with Jest timers in some way.

Here real timers are perfectly fine. Fake timers provide two services and we use neither of them: i) execute timeouts like 5000ms without actually waiting for 5 seconds, by "compressing" time and moving through it manually; ii) be able to stop at a precise moment in time and execute code there.

Reply:
In my previous experience, I seem to recall that relying on real timers could sometimes cause tests to fail with a timeout error — although things may have changed as time passed.

Reply:
Jest allows a test to run up to 5 seconds and then aborts it and reports error. That might be the timeout you've seen. But `userEvent.click` merely does a `setTimeout( continue, 0 )` to do an event loop tick between two input events. With `userEvent.click( { delay: 5000 } )` there would be a longer, configurable delay.

Reply:
I believe that in my case it was more of a glitch in how `user-event`'s calls to `setTimeout` worked, although I don't have much evidence at hand. I just remember that I initially got around that issue by using `delay: null` [as suggested in this GitHub issue](https://github.com/testing-library/user-event/issues/833#issuecomment-1035334908). Later I ended up using [the `advanceTimers` option](https://github.com/testing-library/user-event/pull/907) after it was introduced in version 14.2

Reply:
Yes, `delay: null` works because when it's not a number, the waiting code is skipped:

https://github.com/testing-library/user-event/blob/main/src/utils/misc/wait.ts

The default value of `delay` is `0`, and that leads to a `setTimeout( r, 0 )`. That's confusing because when your project uses fake timers by default, and even when there are no timers at all in the tested code, the `userEvent` internals still get stuck because nobody is advancing the timers for them.